transport/src/main/java/io/netty/channel/AbstractChannel.java

```diff
@@ -96,12 +96,26 @@ protected AbstractChannel(Channel parent, ChannelId id) {
         pipeline = new DefaultChannelPipeline(this);
     }
 
+    protected AbstractChannel(Channel parent, ChannelId id, ChannelInboundHandler pipelineTail) {
+        this.parent = parent;
+        this.id = id;
+        unsafe = newUnsafe();
+        pipeline = new DefaultChannelPipeline(this, pipelineTail);
+    }
+
     @Override
     public final ChannelId id() {
         return id;
     }
 
     /**
+     * If the {@code pipelineTail} was specified in the constructor, this will return that object.
+     */
+    protected ChannelInboundHandler pipelineTail() {
```

Comment:
exposing this method as `protected` is a hack because `EmbeddedChannel` lives in a different package. An option is to deprecate the existing `EmbeddedChannel` and move it into the channel package ... but that seems a bit heavy handed.

@normanmaurer @trustin - Any ideas?